The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/9975858621/job/27566661430) [(plain)](https://github.com/rust-lang/rust/commit/5973845a9c78025694e2ed10cb3eeb9e434e4fd2/checks/27566661430/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Getting action download info
Download action repository 'msys2/setup-msys2@v2.22.0' (SHA:cc11e9188b693c2b100158c3322424c4cc1dadea)
Download action repository 'actions/checkout@v4' (SHA:692973e3d937129bcbf40652eb9f2f61becf3332)
Download action repository 'actions/upload-artifact@v4' (SHA:0b2256b8c012f0828dc542b3febcab082c67f72b)
Complete job name: PR - mingw-check-tidy
git config --global core.autocrlf false
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
---
COPY scripts/sccache.sh /scripts/
RUN sh /scripts/sccache.sh

COPY host-x86_64/mingw-check/reuse-requirements.txt /tmp/
RUN pip3 install --no-deps --no-cache-dir --require-hashes -r /tmp/reuse-requirements.txt \
    && pip3 install virtualenv
COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---

#12 [5/8] COPY host-x86_64/mingw-check/reuse-requirements.txt /tmp/
#12 DONE 0.0s

#13 [6/8] RUN pip3 install --no-deps --no-cache-dir --require-hashes -r /tmp/reuse-requirements.txt     && pip3 install virtualenv
#13 0.460   Downloading binaryornot-0.4.4-py2.py3-none-any.whl (9.0 kB)
#13 0.478 Collecting boolean-py==4.0
#13 0.485   Downloading boolean.py-4.0-py3-none-any.whl (25 kB)
#13 0.504 Collecting chardet==5.1.0
---
#13 3.792 Building wheels for collected packages: reuse
#13 3.793   Building wheel for reuse (pyproject.toml): started
#13 4.143   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 4.144   Created wheel for reuse: filename=reuse-1.1.0-cp310-cp310-manylinux_2_35_x86_64.whl size=181117 sha256=f5f58750481f69515c2c0d1d503daf565e2565c370d07fc6aeb95fe3498b4269
#13 4.144   Stored in directory: /tmp/pip-ephem-wheel-cache-f38is3gj/wheels/c2/3c/b9/1120c2ab4bd82694f7e6f0537dc5b9a085c13e2c69a8d0c76d
#13 4.147 Installing collected packages: boolean-py, binaryornot, setuptools, reuse, python-debian, markupsafe, license-expression, jinja2, chardet
#13 4.172   Attempting uninstall: setuptools
#13 4.172     Found existing installation: setuptools 59.6.0
#13 4.174     Not uninstalling setuptools at /usr/lib/python3/dist-packages, outside environment /usr
#13 4.174     Not uninstalling setuptools at /usr/lib/python3/dist-packages, outside environment /usr
#13 4.174     Can't uninstall 'setuptools'. No files were found to uninstall.
#13 4.896 Successfully installed binaryornot-0.4.4 boolean-py-4.0 chardet-5.1.0 jinja2-3.1.2 license-expression-30.0.0 markupsafe-2.1.1 python-debian-0.1.49 reuse-1.1.0 setuptools-66.0.0
#13 4.896 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#13 5.450 Collecting virtualenv
#13 5.504   Downloading virtualenv-20.26.3-py3-none-any.whl (5.7 MB)
#13 5.731      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 5.7/5.7 MB 25.3 MB/s eta 0:00:00
#13 5.787 Collecting platformdirs<5,>=3.9.1
#13 5.795   Downloading platformdirs-4.2.2-py3-none-any.whl (18 kB)
#13 5.834 Collecting filelock<4,>=3.12.2
#13 5.842   Downloading filelock-3.15.4-py3-none-any.whl (16 kB)
#13 5.863 Collecting distlib<1,>=0.3.7
#13 5.871   Downloading distlib-0.3.8-py2.py3-none-any.whl (468 kB)
#13 5.883      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 468.9/468.9 KB 47.8 MB/s eta 0:00:00
#13 5.973 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 6.165 Successfully installed distlib-0.3.8 filelock-3.15.4 platformdirs-4.2.2 virtualenv-20.26.3
#13 DONE 6.3s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      186304 kB
DirectMap2M:     8202240 kB
DirectMap1G:    10485760 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
    Finished `dev` profile [unoptimized] target(s) in 0.04s
##[endgroup]
downloading https://ci-artifacts.rust-lang.org/rustc-builds-alt/3de0a7c716daab58bdb6551b0d0af6a466486639/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz
extracting /checkout/obj/build/cache/llvm-3de0a7c716daab58bdb6551b0d0af6a466486639-true/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz to /checkout/obj/build/x86_64-unknown-linux-gnu/ci-llvm
---
   Compiling tidy v0.1.0 (/checkout/src/tools/tidy)
    Finished `release` profile [optimized] target(s) in 29.61s
##[endgroup]
fmt check
##[error]Diff in /checkout/compiler/rustc_target/src/spec/targets/x86_64_unknown_none.rs at line 18:
         relro_level: RelroLevel::Full,
         linker_flavor: LinkerFlavor::Gnu(Cc::No, Lld::Yes),
         linker: Some("rust-lld".into()),
-        features:
-            "-mmx,-sse,-sse2,-sse3,-ssse3,-sse4.1,-sse4.2,-avx,-avx2,+soft-float"
-                .into(),
+        features: "-mmx,-sse,-sse2,-sse3,-ssse3,-sse4.1,-sse4.2,-avx,-avx2,+soft-float".into(),
         supported_sanitizers: SanitizerSet::KCFI | SanitizerSet::KERNELADDRESS,
         disable_redzone: true,
         panic_strategy: PanicStrategy::Abort,
fmt error: Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt/bin/rustfmt" "--config-path" "/checkout" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/checkout/compiler/rustc_target/src/spec/targets/armv7_unknown_linux_musleabihf.rs" "/checkout/compiler/rustc_target/src/spec/targets/aarch64_unknown_linux_gnu_ilp32.rs" "/checkout/compiler/rustc_target/src/spec/targets/thumbv7neon_unknown_linux_gnueabihf.rs" "/checkout/compiler/rustc_target/src/spec/targets/aarch64_kmc_solid_asp3.rs" "/checkout/compiler/rustc_target/src/spec/targets/i586_pc_windows_msvc.rs" "/checkout/compiler/rustc_target/src/spec/targets/wasm32_wasip1.rs" "/checkout/compiler/rustc_target/src/spec/targets/thumbv5te_none_eabi.rs" "/checkout/compiler/rustc_target/src/spec/targets/armv7r_none_eabi.rs" "/checkout/compiler/rustc_target/src/spec/targets/aarch64_unknown_teeos.rs" "/checkout/compiler/rustc_target/src/spec/targets/armv7_unknown_linux_uclibceabi.rs" "/checkout/compiler/rustc_target/src/spec/targets/s390x_unknown_linux_musl.rs" "/checkout/compiler/rustc_target/src/spec/targets/armv7a_kmc_solid_asp3_eabihf.rs" "/checkout/compiler/rustc_target/src/spec/targets/aarch64_unknown_freebsd.rs" "/checkout/compiler/rustc_target/src/spec/targets/mips64_unknown_linux_gnuabi64.rs" "/checkout/compiler/rustc_target/src/spec/targets/aarch64_pc_windows_msvc.rs" "/checkout/compiler/rustc_target/src/spec/targets/thumbv7a_pc_windows_msvc.rs" "/checkout/compiler/rustc_target/src/spec/targets/aarch64_unknown_uefi.rs" "/checkout/compiler/rustc_target/src/spec/targets/arm_linux_androideabi.rs" "/checkout/compiler/rustc_target/src/spec/targets/x86_64_win7_windows_msvc.rs" "/checkout/compiler/rustc_target/src/spec/targets/x86_64_unknown_illumos.rs" "/checkout/compiler/rustc_target/src/spec/targets/arm64_32_apple_watchos.rs" "/checkout/compiler/rustc_target/src/spec/targets/csky_unknown_linux_gnuabiv2.rs" "/checkout/compiler/rustc_target/src/spec/targets/i686_unknown_openbsd.rs" "/checkout/compiler/rustc_target/src/spec/targets/mipsisa64r6_unknown_linux_gnuabi64.rs" "/checkout/compiler/rustc_target/src/spec/targets/powerpc_unknown_freebsd.rs" "/checkout/compiler/rustc_target/src/spec/targets/x86_64_apple_tvos.rs" "/checkout/compiler/rustc_target/src/spec/targets/arm64e_apple_ios.rs" "/checkout/compiler/rustc_target/src/spec/targets/riscv64gc_unknown_openbsd.rs" "/checkout/compiler/rustc_target/src/spec/targets/arm64ec_pc_windows_msvc.rs" "/checkout/compiler/rustc_target/src/spec/targets/x86_64_unknown_linux_gnu.rs" "/checkout/compiler/rustc_target/src/spec/targets/armv5te_unknown_linux_gnueabi.rs" "/checkout/compiler/rustc_target/src/spec/targets/thumbv7neon_linux_androideabi.rs" "/checkout/compiler/rustc_target/src/spec/targets/i586_unknown_netbsd.rs" "/checkout/compiler/rustc_target/src/spec/targets/x86_64_uwp_windows_gnu.rs" "/checkout/compiler/rustc_target/src/spec/targets/i686_pc_windows_msvc.rs" "/checkout/compiler/rustc_target/src/spec/targets/powerpc_unknown_linux_gnuspe.rs" "/checkout/compiler/rustc_target/src/spec/targets/armv7a_none_eabihf.rs" "/checkout/compiler/rustc_target/src/spec/targets/i586_unknown_linux_gnu.rs" "/checkout/compiler/rustc_target/src/spec/targets/sparc64_unknown_netbsd.rs" "/checkout/compiler/rustc_target/src/spec/targets/powerpc_unknown_linux_gnu.rs" "/checkout/compiler/rustc_target/src/spec/targets/thumbv8m_base_none_eabi.rs" "/checkout/compiler/rustc_target/src/spec/targets/wasm32_wasi.rs" "/checkout/compiler/rustc_target/src/spec/targets/mipsel_unknown_linux_gnu.rs" "/checkout/compiler/rustc_target/src/spec/targets/arm_unknown_linux_gnueabihf.rs" "/checkout/compiler/rustc_target/src/spec/targets/powerpc64_unknown_openbsd.rs" "/checkout/compiler/rustc_target/src/spec/targets/xtensa_esp32s2_none_elf.rs" "/checkout/compiler/rustc_target/src/spec/targets/riscv32imc_unknown_none_elf.rs" "/checkout/compiler/rustc_target/src/spec/targets/bpfeb_unknown_none.rs" "/checkout/compiler/rustc_target/src/spec/targets/bpfel_unknown_none.rs" "/checkout/compiler/rustc_target/src/spec/targets/powerpc_unknown_netbsd.rs" "/checkout/compiler/rustc_target/src/spec/targets/x86_64_unknown_dragonfly.rs" "/checkout/compiler/rustc_target/src/spec/targets/aarch64_unknown_linux_gnu.rs" "/checkout/compiler/rustc_target/src/spec/targets/riscv64gc_unknown_none_elf.rs" "/checkout/compiler/rustc_target/src/spec/targets/x86_64_unknown_hermit.rs" "/checkout/compiler/rustc_target/src/spec/targets/i686_unknown_linux_musl.rs" "/checkout/compiler/rustc_target/src/spec/targets/armv8r_none_eabihf.rs" "/checkout/compiler/rustc_target/src/spec/targets/x86_64_uwp_windows_msvc.rs" "/checkout/compiler/rustc_target/src/spec/targets/powerpc64_ibm_aix.rs" "/checkout/compiler/rustc_target/src/spec/targets/i686_unknown_freebsd.rs" "/checkout/compiler/rustc_target/src/spec/targets/armv7a_none_eabi.rs" "/checkout/compiler/rustc_target/src/spec/targets/loongarch64_unknown_linux_musl.rs" "/checkout/compiler/rustc_target/src/spec/targets/hexagon_unknown_linux_musl.rs" "/checkout/compiler/rustc_target/src/spec/targets/x86_64_unknown_none.rs" "/checkout/compiler/rustc_mir_build/src/check_unsafety.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
  local time: Wed Jul 17 14:18:27 UTC 2024
  network time: Wed, 17 Jul 2024 14:18:27 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/compiler/rustc_target/src/spec/targets/x86_64_unknown_none.rs
+++ b/compiler/rustc_target/src/spec/targets/x86_64_unknown_none.rs
@@ -18,9 +18,7 @@ pub fn target() -> Target {
         relro_level: RelroLevel::Full,
         linker_flavor: LinkerFlavor::Gnu(Cc::No, Lld::Yes),
         linker: Some("rust-lld".into()),
-        features:
-            "-mmx,-sse,-sse2,-sse3,-ssse3,-sse4.1,-sse4.2,-3dnow,-3dnowa,-avx,-avx2,+soft-float"
-                .into(),
+        features: "-mmx,-sse,-sse2,-sse3,-ssse3,-sse4.1,-sse4.2,-avx,-avx2,+soft-float".into(),
         supported_sanitizers: SanitizerSet::KCFI | SanitizerSet::KERNELADDRESS,
         disable_redzone: true,
         panic_strategy: PanicStrategy::Abort,

